File: android/src/main/java/io/plugins/admob_platform_view/AdmobPlatformViewPlugin.java
package io.plugins.admob_platform_view;

import com.google.android.gms.ads.MobileAds;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;
import io.flutter.plugin.common.PluginRegistry.Registrar;

/** AdmobPlatformViewPlugin */
public class AdmobPlatformViewPlugin implements MethodCallHandler {
  /** Plugin registration. */
  public static void registerWith(Registrar registrar) {
    registrar
        .platformViewRegistry()
        .registerViewFactory(
            "plugins.flutter.io/firebase_admob/adview", new FlutterAdViewFactory());

    final MethodChannel channel = new MethodChannel(registrar.messenger(), "plugins.flutter.io/firebase_admob");
    channel.setMethodCallHandler(new AdmobPlatformViewPlugin(registrar));
  }

  final Registrar registrar;

  private AdmobPlatformViewPlugin(Registrar registrar) {
    this.registrar = registrar;
  }

  @Override
  public void onMethodCall(MethodCall call, Result result) {
    if (call.method.equals("initialize")) {
      callInitialize(call, result);
    } else {
      result.notImplemented();
    }
  }

  private void callInitialize(MethodCall call, Result result) {
    String appId = call.argument("appId");
    if (appId == null || appId.isEmpty()) {
      result.error("no_app_id", "a null or empty AdMob appId was provided", null);
      return;
    }

    MobileAds.initialize(registrar.context(), appId);
    result.success(null);
  }
}

File: ios/Classes/AdmobPlatformViewPlugin.m
#import "AdmobPlatformViewPlugin.h"
#import "Firebase.h"

@implementation AdmobPlatformViewPlugin
- (instancetype)init {
  self = [super init];
  if (self && ![FIRApp appNamed:@"__FIRAPP_DEFAULT"]) {
    NSLog(@"Configuring the default Firebase app...");
    [FIRApp configure];
    NSLog(@"Configured the default Firebase app %@.", [FIRApp defaultApp].name);
  }
  return self;
}

+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  [registrar registerViewFactory:[[FLTAdViewFactory alloc] init] withId:@"plugins.flutter.io/firebase_admob/adview"];
  [registrar registerViewFactory:[[FLTNativeAdViewFactory alloc] init] withId:@"plugins.flutter.io/firebase_admob/adview/native"];

  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"plugins.flutter.io/firebase_admob"
            binaryMessenger:[registrar messenger]];

  AdmobPlatformViewPlugin* instance = [[AdmobPlatformViewPlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([call.method isEqualToString:@"initialize"]) {
    [self callInitialize:call result:result];
  } else {
    result(FlutterMethodNotImplemented);
  }
}

- (void)callInitialize:(FlutterMethodCall *)call result:(FlutterResult)result {
  NSString *appId = (NSString *)call.arguments[@"appId"];
  if (appId == nil || [appId length] == 0) {
    result([FlutterError errorWithCode:@"no_app_id"
                               message:@"a non-empty AdMob appId was not provided"
                               details:nil]);
    return;
  }

  [GADMobileAds configureWithApplicationID:appId];
  result(nil);
}
@end
